Plots/Contours/NCL_polar_1_lg.py

```diff
@@ -0,0 +1,115 @@
+
+"""
+NCL_polar_1_lg.py
+================
+
+This script illustrates the following concepts:
+    - Drawing black-and-white contours over a polar stereographic map
+    - Drawing the northern hemisphere of a polar stereographic map
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/polar_1_lg.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/polar_1_lg.png
+"""
+###############################################################################
+# Import packages:
+import numpy as np
+import xarray as xr
+import cartopy.feature as cfeature
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import matplotlib.path as mpath
+import geocat.datafiles as gdf
+import matplotlib.ticker as mticker
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/uv300.nc"))
+U = ds.U[1, :, :]
+
+###############################################################################
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+wrap_U = gvutil.xr_add_cyclic_longitudes(U, "lon")
+
+###############################################################################
+#Plot:
+    
+# Generate figure (set its size (width, height) in inches)
+fig = plt.figure(figsize=(10, 10))
+
+# Generate axes, using Cartopy, drawing coastlines, and adding features
+fig = plt.figure(figsize=(10, 10))
+projection = ccrs.NorthPolarStereo()
+ax = plt.axes(projection=projection)
+ax.coastlines(linewidths=0.5)
+ax.add_feature(cfeature.LAND, facecolor='lightgray')
+
+# Set extent to include latitudes between 0 and 40 and longitudes between
+# -180 and 180 only
+ax.set_extent([-180, 180, 0, 40], ccrs.PlateCarree())
+
+# Set draw_labels to False so that you can manually manipulate it later
+gl = ax.gridlines(
+    ccrs.PlateCarree(),
+    draw_labels=False,
+    linestyle="--",
+    color='k')
+
+# Manipulate latitude and longitude gridline numbers and spacing
+gl.ylocator = mticker.FixedLocator(np.arange(0, 90, 15))
+gl.xlocator = mticker.FixedLocator(np.arange(-180, 180, 30))
+
+# Set boundary to a circle
+theta = np.linspace(0, 2 * np.pi, 100)
+center, radius = [0.5, 0.5], 0.5
+verts = np.vstack([np.sin(theta), np.cos(theta)]).T
+circle = mpath.Path(verts * radius + center)
+ax.set_boundary(circle, transform=ax.transAxes)
+
+# Manipulate longitude labels (0, 30 E, 60 E, ..., 30 W, etc.)
+ticks = np.arange(0, 210, 30)
+etick = ['0'] + [r'%d$^\circ$E' %
+                 tick for tick in ticks if (tick != 0) & (tick != 180)] + ['180']
+wtick = [
```

Comment:
@erogluorhan do you think we should have @michaelavs turn this code into a geocat-viz convenience function for circular plots? Thanks!

Reply:
I think that sounds reasonable to me.  Seems like it will be pretty standard, though maybe it makes sense to wait on that decision until we actually see duplication?

Reply:
@clyne @kmpalul we should definitely think of adding such a solution for circular plots in **geocat-viz**; however, this code includes pretty much custom (i.e. hard-coded) numeric values in its current version, which might need to be changed for other cases or be calculated in a generic way with respect to plot size etc. For example, two arrays for tick label locations:

```
xticks = [-0.8, 28, 58, 89.1, 120, 151, 182.9, -36, -63, -89, -114, -140]
yticks = [-3] + [-2] + [-1] + [-1] * 2 + [-1] + [-3] + [-7] + [-7] * 3 + [-7]
```

So, my recommendation is that @michaelavs could try to come up with a generic calculation for `xtick, ytick` locations in this case. If that's not easy on a single case, we may want to study on a few of additional similar plotting examples that need this exact solution then come up with a generic calculation. 

Does that make sense?

Reply:
Sounds good to me.